packages/playwright-core/src/server/browserType.ts

```diff
@@ -38,12 +38,18 @@ import { helper } from './helper';
 import { RecentLogsCollector } from '../utils/debugLogger';
 import type { CallMetadata } from './instrumentation';
 import { SdkObject } from './instrumentation';
-import { ManualPromise } from '../utils/manualPromise';
 import { type ProtocolError, isProtocolError } from './protocolError';
 
 export const kNoXServerRunningError = 'Looks like you launched a headed browser without having a XServer running.\n' +
   'Set either \'headless: true\' or use \'xvfb-run <your-playwright-app>\' before running Playwright.\n\n<3 Playwright Team';
 
+
+export interface BrowserReadyState {
+  onBrowserOutput(message: string): void;
+  onBrowserExit(): void;
+  ready(): Promise<string|undefined>;
```

Comment:
I'd make this a bit more verbose:

```suggestion
  waitUntilReady(): Promise<{ wsEndpoint?: string }>;
```